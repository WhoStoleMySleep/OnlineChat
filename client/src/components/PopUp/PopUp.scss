.pop-up {
  background: rgba(255, 255, 255, 0.5);
  border-radius: 4px;
  bottom: 40px;
  max-height: 60%;
  overflow: hidden;
  position: absolute;
  right: 40px;
  width: 546px;
  z-index: 999999999;

  &.hidden {
    display: none;
  }

  &__close {
    background: none;
    border: none;
    font-size: 25px;
    line-height: 17px;
    outline: none;
    padding: 0;
    position: absolute;
    right: 10px;
    top: 10px;
    transition-duration: 500ms;
    width: 20px;

    &:hover {
      transform: rotate(180deg);
    }
  }

  &__list {
    font-size: 18px;
    padding: 0 40px;
  }

  &__item {
    word-wrap: break-word;
    margin-top: 10px;
  }
}

@media all and (max-width: 768px) {
  .pop-up {
    height: 100vh;
    max-height: none;
    padding-top: 40px;
    right: auto;
    top: 0;
    width: 100%;

    &__close {
      top: 40px;
    }
  }
}
