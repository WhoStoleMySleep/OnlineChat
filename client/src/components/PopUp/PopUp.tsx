import React from 'react';
import { useSelector } from 'react-redux';
import useClassHidden from '../../hooks/useClosePopUp/useClosePopUp';
import { setUnreadMessages } from '../../redux/componentReducers/unreadMessages';
import styles from './PopUp.module.scss';

function PopUp(props: { headed: string }) {
  const { headed } = props;

  const { unreadMessages } = useSelector(
    (state: { unreadMessages: { unreadMessages: any; }; }) =>
      state.unreadMessages
  );

  const { onClick: onPopUpClose } = useClassHidden(
    styles['pop-up'],
    styles.hidden,
    setUnreadMessages
  );

  return (
    <div className={`${styles['pop-up']} ${unreadMessages.length ? '' : styles.hidden}`}>
      <button
        type="button"
        className={styles['pop-up__close']}
        onClick={onPopUpClose}
      >
        &#215;
      </button>
      <h2 className={styles['pop-up__headed']}>{headed}</h2>
      <ol className={styles['pop-up__list']}>
        {unreadMessages.length
          && unreadMessages.map((res: { text: string; id: string }) => (
            <li className={styles['pop-up__item']} key={res.id}>
              {res.text}
            </li>
          ))}
      </ol>
    </div>
  );
}

export default PopUp;
