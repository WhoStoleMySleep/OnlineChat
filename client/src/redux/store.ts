import { configureStore } from '@reduxjs/toolkit';
import author from './componentReducers/login';
import messages from './componentReducers/messages';

export default configureStore({
  reducer: {
    messages,
    author
  }
}); 