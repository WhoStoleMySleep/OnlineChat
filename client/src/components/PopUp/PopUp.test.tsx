import '@testing-library/jest-dom';
import {
  render, screen
} from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import PopUp from './PopUp';
import store from '../../redux/store';

describe('Message', () => {
  it('Message snapshot', async () => {
    render(
      <Provider store={store}>
        <PopUp headed="New messages!" />
      </Provider>
    );

    const popUp = screen.getByTestId('popUp');

    expect(popUp).toMatchSnapshot();
  });
});
