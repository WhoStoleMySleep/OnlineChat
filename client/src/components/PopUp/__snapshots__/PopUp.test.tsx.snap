// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message snapshot 1`] = `
<div
  class="pop-up"
  data-testid="popUp"
>
  <button
    class="pop-up__close"
    type="button"
  >
    Ã—
  </button>
  <h2
    class="pop-up__headed"
  >
    New messages!
  </h2>
  <ol
    class="pop-up__list"
  >
    0
  </ol>
</div>
`;
